src/operator/tensor/matrix_op-inl.h

```diff
@@ -257,6 +257,29 @@ struct TransposeParam : public dmlc::Parameter<TransposeParam> {
   }
 };
 
+
+template<typename xpu, typename DType>
+MSHADOW_XINLINE void Transpose2D(DType *in, DType *out, index_t shape_0, index_t shape_1) {
+// ensure cache line hits and prevent cache miss for any configuration
+index_t blocksize = 32;
+index_t n = shape_0;
+index_t p = shape_1;
+
+for (index_t i = 0; i < n; i += blocksize) {
```

Comment:
How is this blocksize decided? Is it platform dependent?

Reply:
Gave an explanation in the description and function definition.